src/Entity/SmsMessage.php

```diff
@@ -34,11 +36,11 @@
 class SmsMessage extends ContentEntityBase implements SmsMessageInterface {
 
   /**
-   * The result associated with this SMS message.
+   * Temporarily stores the message result until save().
    *
-   * @var \Drupal\sms\Message\SmsMessageResultInterface|NULL
+   * @var \Drupal\sms\Message\SmsMessageResultInterface|\Drupal\sms\Entity\SmsMessageResultInterface|null
    */
-  protected $result;
+  protected $result = NULL;
```

Comment:
following on from below entity freshness comments, this variable should effectively just be **unsaved** result. It shouldnt be a cache of saved results. This variable should be dumped as soon as the message is saved.